packages/xrpl/src/models/transactions/NFTokenMint.ts

```diff
@@ -140,4 +165,16 @@ export function validateNFTokenMint(tx: Record<string, unknown>): void {
   if (tx.NFTokenTaxon == null) {
     throw new ValidationError('NFTokenMint: missing field NFTokenTaxon')
   }
+
```

Comment:
```suggestion
  if (!((tx.Amount == null && tx.Expiration == null && tx.Destination == null) || (tx.Amount != null && tx.Expiration != null && tx.Destination != null))) {
    if (tx.Expiration != null || tx.Destination != null) {
      throw new ValidationError(
        'NFTokenMint: All three fields Amount, Expiration, Destination must be present',
      )
    }
  }
```

We will need all three fields to be present (or) absent to make logical sense, isn't it? Isn't this a more comprehensive check?

I'm guessing this is already checked in rippled, so perhaps there is no need for a rigorous check here.

Reply:
line 169 seems unnecessary as well

Reply:
`Expiration` and `Destination` can only be set if `Amount` is present, but they are still optional fields.

Reply:
What happens if a developer specifies `Destination` field but not the `Amount` field?

Reply:
it will be `temMALFORMED`

Reply:
Yes, but that error is returned from the `rippled` node. I feel we can catch that error inside client-libraries itself.

However, this is just a suggestion. I like the PR even without this change.